Platon/backend/app/search_engine/views.py

```diff
@@ -0,0 +1,192 @@
+                            result_id_score.append((user.id,score))
+                        else:
+                            id_index = id_list.index(user[0])
+                            result_id_score[id_index] =(user.id, result_id_score[id_index][1]+score)
+                except:
+                    return make_response(jsonify({"error": "Database Connection Problem."}), 500)
+            # Sort result ids according to their scores
+            sorted_id_list = SearchEngine.sort_ids(result_id_score)
+            sorted_result_list = []
+            for user_id,score in sorted_id_list:
+                index = [user["id"] for user in result_list].index(user_id)
+                sorted_result_list.append(result_list[index])
+            # Apply given filters
+            if form.job_filter.data is not None:
+                for i,user in enumerate(sorted_result_list):
+                    if user["job_id"] != form.job_filter.data:
+                        sorted_result_list.pop(i)
+            number_of_pages = 1
+            # Apply Pagination
+            if form.page.data is not None and form.per_page.data:
+                per_page = form.per_page.data
+                number_of_pages = math.ceil(len(sorted_id_list)/per_page)
+                page = form.page.data if form.page.data < number_of_pages else number_of_pages-1
+                sorted_result_list = sorted_result_list[page*per_page:(page+1)*per_page]
+            # Add Search History Item
+            try:
+                auth_token = request.headers.get('auth_token')
+                SearchEngine.add_search_history_item(decode_token(auth_token),search_query,int(SearchType.USER))
+            except:
+                pass
+            return make_response(jsonify({"number_of_pages": number_of_pages,"result_list": sorted_result_list}))
+        return make_response(jsonify({"error": "Missing data fields or invalid data."}), 400)
+
+@search_engine_ns.route("/search_history")
+class SearchHistoryAPI(Resource):
+    
+    @api.doc(responses={401 : 'Account Problems', 400 : 'Input Format Error' ,500 : ' Database Connection Error'})
+    @api.response(200, 'Valid Search History', search_history_list_model)
+    @api.expect(search_history_parser)
+    @login_required
+    def get(user_id,self):
+        form = SearchHistoryForm(request.args)
+        if form.validate():
+            if form.search_type.data > 2 or form.search_type.data < 0:
+                return make_response(jsonify({"error": "Search Type must be 0, 1 or 2"}), 400)
+            try:
+                search_history =  db.engine.execute("SELECT `query` FROM `search_history_item` WHERE `user_id`={} AND `type`={}".format(user_id,form.search_type.data))           
+            except:
+                return make_response(jsonify({"error": "Database Connection Problem."}), 500)
+            search_hist_dict = {}
+            for history_item in search_history:
```

Comment:
Is there any specific advantages of sorting search history with respect to number of times the query performed?

Reply:
Think about a case a user searches for same query for so many times. For these cases I have to remove duplicates. So , while removing duplicates, I also sorting them according the their usaga rate. There is not any specification about them in requirements. So I made the decission.